### PR TITLE
Closes ita-social-projects/Car-Back-End/issues/650

### PR DESCRIPTION
**Pull request by vladzavorotny**

### CODE REVIEW

**Review by AndriiDzendzia — APPROVED** [review]

**Review by Vlas2001 — APPROVED** [review]

File: Car.Backend/Car.Domain/Services/Implementation/JourneyService.cs [hunk 1]
```diff
@@ -324,13 +327,17 @@ public async Task<bool> DeleteUserFromJourney(int journeyId, int userId)
             return true;
         }
 
-        public async Task<bool> AddUserToJourney(int journeyId, int userId, IEnumerable<StopDto> applicantStops)
+        public async Task<bool> AddUserToJourney(JourneyApplyModel journeyApply)
         {
+            var journeyId = journeyApply.JourneyUser?.JourneyId ?? default(int);
```
Comment (serhii-zinkiv) [inline Car.Backend/Car.Domain/Services/Implementation/JourneyService.cs:332]: default(int) ? why did you use it?

File: Car.Backend/Car.Domain/Services/Implementation/JourneyService.cs [hunk 2]
```diff
@@ -324,13 +327,17 @@ public async Task<bool> DeleteUserFromJourney(int journeyId, int userId)
...
+
             var journey = await journeyRepository
                 .Query()
                 .IncludeAllParticipants()
                 .FirstOrDefaultAsync(j => j.Id == journeyId);
 
+            var userId = journeyApply.JourneyUser?.UserId ?? default(int);
```
Comment (serhii-zinkiv) [inline Car.Backend/Car.Domain/Services/Implementation/JourneyService.cs:339]: same

File: Car.Backend/Car.Domain/Services/Implementation/JourneyService.cs [hunk 3]
```diff
@@ -355,10 +362,23 @@ public async Task<bool> AddUserToJourney(int journeyId, int userId, IEnumerable<
             }
 
             await journeyRepository.SaveChangesAsync();
+            if (journeyApply.JourneyUser is not null)
```
Comment (serhii-zinkiv) [inline Car.Backend/Car.Domain/Services/Implementation/JourneyService.cs:365]: can you use it here != operator?

File: Car.Backend/Car.Domain/Services/Implementation/JourneyUserService.cs [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+        public JourneyUserService(
+            IRepository<JourneyUser> journeyUserRepository,
+            IMapper mapper)
+        {
+            this.journeyUserRepository = journeyUserRepository;
+            this.mapper = mapper;
+        }
+
+        public async Task<JourneyUserDto> GetJourneyUserByIdAsync(int journeyId, int userId)
```
Comment (serhii-zinkiv) [inline Car.Backend/Car.Domain/Services/Implementation/JourneyUserService.cs:28]: Can you use here userId from context?

File: Car.Backend/Car.Domain/Services/Implementation/JourneyUserService.cs [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        {
+            var journeyUser = await journeyUserRepository
+                .Query()
+                .FirstOrDefaultAsync(ju => ju.JourneyId == journeyId && ju.UserId == userId);
+
+            return mapper.Map<JourneyUser, JourneyUserDto>(journeyUser);
+        }
+
+        public async Task<bool> HasBaggage(int journeyId, int userId)
```
Comment (serhii-zinkiv) [inline Car.Backend/Car.Domain/Services/Implementation/JourneyUserService.cs:37]: same

File: Car.Backend/Car.Domain/Services/Implementation/JourneyUserService.cs [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+                journeyUserRepository.Detach(journeyUser);
+                journeyUser = await journeyUserRepository.UpdateAsync(updatedJourneyUser);
+                await journeyUserRepository.SaveChangesAsync();
+                return mapper.Map<JourneyUser, JourneyUserDto>(journeyUser);
+            }
+
+            return null;
+        }
+
+        public async Task<JourneyUserDto?> SetWithBaggageAsync(int journeyId, int userId, bool withBaggage)
```
Comment (serhii-zinkiv) [inline Car.Backend/Car.Domain/Services/Implementation/JourneyUserService.cs:65]: same

**Pull request merged**